ext/serializer.c

```diff
@@ -856,9 +856,21 @@ static void _serialize_meta(zval *el, ddtrace_span_data *span) {
 
     zend_array *span_links = ddtrace_spandata_property_links(span);
     if (zend_hash_num_elements(span_links) > 0) {
+        // Save the current exception, if any, and clear it for php_json_encode_serializable_object not to fail
+        // and zend_call_function to actually call the jsonSerialize method
+        // Restored after span links are serialized
+        zend_object* current_exception = NULL;
+        if (EG(exception)) {
+            current_exception = EG(exception);
+            EG(exception) = NULL;
+        }
```

Comment:
```suggestion
        zend_object* current_exception = EG(exception);
        EG(exception) = NULL;
```
you can remove the condition and execute that unconditionally